datasets/new_york_taxi_trips/pipelines/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,422 @@
+    )
+    client = bigquery.Client(project=project_id)
+    table_ref = client.dataset(dataset_id).table(table_id)
+    job_config = bigquery.LoadJobConfig()
+    job_config.source_format = bigquery.SourceFormat.CSV
+    job_config.skip_leading_rows = 1  # ignore the header
+    job_config.autodetect = False
+    with open(file_path, "rb") as source_file:
+        job = client.load_table_from_file(source_file, table_ref, job_config=job_config)
+    job.result()
+    logging.info(
+        f"Loading data from {file_path} into {project_id}.{dataset_id}.{table_id} completed"
+    )
+
+
+def create_dest_table(
+    project_id: str,
+    dataset_id: str,
+    table_id: str,
+    schema_filepath: list,
+    bucket_name: str,
+) -> bool:
+    table_ref = f"{project_id}.{dataset_id}.{table_id}"
+    logging.info(f"Attempting to create table {table_ref} if it doesn't already exist")
+    client = bigquery.Client()
+    success = False
+    try:
+        table_exists_id = client.get_table(table_ref).table_id
+        logging.info(f"Table {table_exists_id} currently exists.")
+        success = True
+    except NotFound:
+        logging.info(
+            (
+                f"Table {table_ref} currently does not exist.  Attempting to create table."
+            )
+        )
+        schema = create_table_schema([], bucket_name, schema_filepath)
+        table = bigquery.Table(table_ref, schema=schema)
+        client.create_table(table)
+        print(f"Table {table_ref} was created".format(table_id))
+        success = True
+    return success
+
+
+def create_table_schema(
+    schema_structure: list, bucket_name: str = "", schema_filepath: str = ""
+) -> list:
+    logging.info(f"Defining table schema... {bucket_name} ... {schema_filepath}")
+    schema = []
+    if not (schema_filepath):
+        schema_struct = schema_structure
+    else:
+        storage_client = storage.Client()
+        bucket = storage_client.get_bucket(bucket_name)
+        blob = bucket.blob(schema_filepath)
+        schema_struct = json.loads(blob.download_as_string(client=None))
+    for schema_field in schema_struct:
+        fld_name = schema_field["name"]
+        fld_type = schema_field["type"]
+        try:
+            fld_descr = schema_field["description"]
+        except KeyError:
+            fld_descr = ""
+        fld_mode = schema_field["mode"]
+        schema.append(
+            bigquery.SchemaField(
+                name=fld_name, field_type=fld_type, mode=fld_mode, description=fld_descr
+            )
+        )
+    return schema
+
+
+def process_month(
+    source_url: str,
+    year_number: int,
+    month_number: int,
+    source_file: str,
+    target_file: str,
+    target_file_name: str,
+    chunksize: str,
+    input_headers: typing.List[str],
+    data_dtypes: dict,
+    output_headers: typing.List[str],
+    pipeline_name: str,
+) -> None:
+    process_month = str(year_number) + "-" + str(month_number).zfill(2)
```

Comment:
fixed

Reply:
@happyhuman - it turns out that this change is not valid because month_number is int when f"{month_number.zfill(2)}" is executed, so it attempts to zfill an int which is incorrect against an int.  Therefore, in order to be able to use your intended syntax it would need to convert month_number to a string prior to executing this line of code.  Therefore the change is moot and problemmatic.  Do you want me to revert back to the original code, or add another line that converts the type prior to executing f"{year_number}-{month_number.zfill(2)}"  ?  Thanks.

Reply:
[2022-02-18 23:45:35,894] {pod_manager.py:197} INFO -     process_month = f"{year_number}-{month_number.zfill(2)}"
[2022-02-18 23:45:35,895] {pod_manager.py:197} INFO - AttributeError: 'int' object has no attribute 'zfill'